src/helpers/ApiServiceHelper.js

```diff
@@ -29,75 +30,275 @@ function ApiServiceHelper (url = '', clientId = '', clientSecret = '', scope = '
       this.clientSecret === '' || this.scope === '') ? false : true;
   };
 
-  this.generateRecordApiUrlsArray = (records, baseApiUrl) => {
-    const functionName = 'setItemApiUrlToRecord';
+  this.processGetItemDataRequests = (itemsArray, token) => {
+    const functionName = 'processGetItemDataRequests';
+
+    return new Promise((resolve, reject) => {
+      async.map(itemsArray, (urlItem, cb) => {
+        return axios.get(urlItem.itemApi, this.constructApiHeaders(token))
+        .then(response => {
+          if (response.data && response.data.data) {
+            cb(null, response.data.data);
+          }
+        })
+        .catch((error) => cb(error));
+      }, (err, results) => {
+        if (err) {
+          let errorResponse = {
+            type: 'item-service-api',
+            function: functionName
+          };
+
+          if (err.response) {
+            // The request was made and the server responded with a status code
+            // that falls out of the range of 2xx
+            errorResponse = HoldRequestConsumerError(Object.assign({}, errorResponse, {
+              message: 'received a status outside of the 2xx range from Item Service',
+              status: err.response.status,
+              error : {
+                method: err.response.request.method,
+                url: err.response.request.path,
+                headers: err.response.headers,
+                data: err.response.data
+              }
+            }));
+          } else if (err.request) {
```

Comment:
In what case does this happen? When we get a 404? Why is the error code set to 500?